implementation/gpioint.tex

```diff
@@ -0,0 +1,179 @@
+\subsection{Porting GPIO Interrupt Driver}
+
+This section considers a case study of an issue discovered when porting the GPIO interrupt driver from SiliconLabs to Rust.
+The issue was discoverd when annotating the \texttt{unsafe} blocks for referencing mutable global state and gives an example of the awareness the inclusion of the \texttt{unsafe} keyword provides.
+
+\subsubsection{Presenting the Problem}
+
+The driver lets the user registers a callback function to be called when an interrupt occures at a given gpio pin.
+It is implemented by a global mutable list of 16 function pointers, a register function to assign functions to indices of the list corresponding to GPIO pins and a dispatch mechanisms which calls the correct functions when an interrupt occures.
+The issue arises in the dispatch function in \autoref{lst:irq-dispatch:c}.
```

Comment:
Avsnittet er bare en linje.

Denne linjen kan kanskje flyttes ned til neste avsnitt, og tror ikke det er ikke nødvendig å refere figuren to ganger i begge setningene heller?